shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -214,6 +214,14 @@ static UITextContentType ToUITextContentType(NSArray<NSString*>* hints) {
   return autofill == nil ? nil : autofill[@"uniqueIdentifier"];
 }
 
+// Read a signed integer from state into an unsigned integer. This is needed for
+// NSRange locations, which are unsigned integers, but are read in from signed
+// integers in state. These are -1 when there is no selection specified, but
+// they should be interpreted as 0 instead of overflowing.
+static NSUInteger readInt(NSInteger value) {
```

Comment:
MIN and MAX appear to cast their inputs to unsigned integers, which is the cause of this bug.  I was seeing things like `MAX(-1, 0) => 2,147,483,647`, and `MIN(-1, 0) => 0`. This function lets us be sure we're sending sensible unsigned ints to MIN and MAX.

Reply:
Hmm maybe that's because `-1` and `0` are number literals, so they got cast to `NSUInteger` before the comparison took place? I think `MAX` should work for if you're converting the number from a `NSNumber` using `intValue`.

Reply:
Here's a specific example I was able to get while running the app.  I saw it at this line:

https://github.com/flutter/engine/blob/3c1e336222fdcb616c4ed3c60119627894abfbeb/shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm#L376

I'm logging like this:

```objectivec
NSLog(@"MAX(%li, 0) => %li, MIN(MAX(range.location, 0), %li) => %li", range.location, MAX(range.location, 0), text.length, MIN(MAX(range.location, 0), text.length));
```

When `range.location` is -1 and `text.length` is 2 this is what I see logged:

```
MAX(-1, 0) => -1, MIN(MAX(range.location, 0), 2) => 2
```

But I would expect the MAX to return 0 and the overall MIN to return 0 as well.